db/src/bootstrap.rs

```diff
@@ -0,0 +1,210 @@
+         (":db.type/long",         25),
+         (":db.type/double",       26),
+         (":db.type/string",       27),
+         (":db.type/boolean",      28),
+         (":db.type/instant",      29),
+         (":db.type/bytes",        30),
+         (":db.cardinality/one",   31),
+         (":db.cardinality/many",  32),
+         (":db.unique/value",      33),
+         (":db.unique/identity",   34),
+         (":db/doc",               35),
+        ]
+    };
+
+    static ref V2_IDENTS: [(&'static str, i64); 2] = {
+        [(":db.schema/version",    36),
+         (":db.schema/attribute",  37),
+        ]
+    };
+
+    static ref V1_PARTS: [(&'static str, i64, i64); 3] = {
+        [(":db.part/db", 0, (1 + V1_IDENTS.len() + V2_IDENTS.len()) as i64),
+         (":db.part/user", 0x10000, 0x10000),
+         (":db.part/tx", 0x10000000, 0x10000000),
+         ]
+    };
+
+    static ref V2_PARTS: [(&'static str, i64, i64); 0] = {
+        []
+    };
+
+    static ref V1_SYMBOLIC_SCHEMA: Value = {
+        let s = r#"
+{:db/ident             {:db/valueType   :db.type/keyword
+                       :db/cardinality :db.cardinality/one
+                       :db/unique      :db.unique/identity}
+ :db.install/partition {:db/valueType   :db.type/ref
+                        :db/cardinality :db.cardinality/many}
+ :db.install/valueType {:db/valueType   :db.type/ref
+                        :db/cardinality :db.cardinality/many}
+ :db.install/attribute {:db/valueType   :db.type/ref
+                        :db/cardinality :db.cardinality/many}
+ ;; TODO: support user-specified functions in the future.
+ ;; :db.install/function {:db/valueType :db.type/ref
+ ;;                       :db/cardinality :db.cardinality/many}
+ :db/txInstant         {:db/valueType   :db.type/long
+                        :db/cardinality :db.cardinality/one
+                        :db/index       true}
+ :db/valueType         {:db/valueType   :db.type/ref
+                        :db/cardinality :db.cardinality/one}
+ :db/cardinality       {:db/valueType   :db.type/ref
+                        :db/cardinality :db.cardinality/one}
+ :db/doc               {:db/valueType   :db.type/string
+                        :db/cardinality :db.cardinality/one}
+ :db/unique            {:db/valueType   :db.type/ref
+                        :db/cardinality :db.cardinality/one}
+ :db/isComponent       {:db/valueType   :db.type/boolean
+                        :db/cardinality :db.cardinality/one}
+ :db/index             {:db/valueType   :db.type/boolean
+                        :db/cardinality :db.cardinality/one}
+ :db/fulltext          {:db/valueType   :db.type/boolean
+                        :db/cardinality :db.cardinality/one}
+ :db/noHistory         {:db/valueType   :db.type/boolean
+                        :db/cardinality :db.cardinality/one}}"#;
+        edn::parse::value(s)
+            .map_err(|_| ErrorKind::BadBootstrapDefinition("Unable to parse V1_SYMBOLIC_SCHEMA".into()))
```

Comment:
I imagine it must have both a runtime and a code-size penalty, since there are instructions involved.

However, Rust in general (and `unwrap` in particular) provides almost no backtrace help, so I think it's worth including until we prove otherwise.